src/status_im2/contexts/emoji_picker/view.cljs

```diff
@@ -0,0 +1,200 @@
+   (i18n/label title)])
+
+(defn- emoji-item
+  [{:keys [unicode] :as emoji} col-index callback close]
+  (let [on-press          (fn []
+                            (when callback
+                              (callback unicode emoji))
+                            (close))
+        last-item-on-row? (= (inc col-index) constants/emojis-per-row)]
+    (fn []
+      [rn/pressable
+       {:style    (style/emoji-container last-item-on-row?)
+        :on-press on-press}
+       [rn/text
+        {:style                    {:font-size constants/emoji-size}
+         :adjusts-font-size-to-fit true
+         :allow-font-scaling       false}
+        unicode]])))
+
+(defn- emoji-row
+  [row-data _ _ {:keys [callback close]}]
+  (into [rn/view {:style style/emoji-row-container}]
+        (map-indexed
+         (fn [col-index {:keys [hexcode] :as emoji}]
+           ^{:key (str hexcode col-index)}
+           [emoji-item emoji col-index callback close])
+         row-data)))
+
+(defn- render-item
+  [item index separator render-data]
+  (if (:header? item)
+    [section-header item index separator render-data]
+    [emoji-row item index separator render-data]))
+
+(defn- empty-result
+  []
+  [quo/empty-state
+   {:title           (i18n/label :t/emoji-no-results-title)
+    :description     (i18n/label :t/emoji-no-results-description)
+    :placeholder?    true
+    :container-style style/empty-results}])
+
+(defn- render-list
+  [{:keys [theme filtered-data on-viewable-items-changed scroll-enabled on-scroll on-select
+           set-scroll-ref close]}]
+  (let [data (if filtered-data filtered-data emoji-picker.data/flatten-data)]
+    [gesture/flat-list
+     {:ref                             set-scroll-ref
+      :scroll-enabled                  @scroll-enabled
+      :data                            data
+      :initial-num-to-render           15
+      :render-fn                       render-item
+      :get-item-layout                 get-item-layout
+      :keyboard-dismiss-mode           :on-drag
+      :keyboard-should-persist-taps    :handled
+      :shows-vertical-scroll-indicator false
+      :on-scroll-to-index-failed       identity
+      :empty-component                 [empty-result]
+      :on-scroll                       on-scroll
+      :render-data                     {:close    close
+                                        :theme    theme
+                                        :callback on-select}
+      :content-container-style         style/list-container-style
+      :viewability-config              {:wait-for-interaction           true
+                                        :item-visible-percent-threshold 100
+                                        :minimum-view-time              200}
+      :on-viewable-items-changed       on-viewable-items-changed}]))
+
+(defn- footer
+  [theme active-category scroll-ref]
+  (let [on-press #(on-press-category %1 %2 active-category scroll-ref)]
```

Comment:
Maybe use `fn` and named arguments to improve readability instead of `%1` and `%2`